cassandra-analytics-core/src/test/java/org/apache/cassandra/spark/bulkwriter/coordinatedwrite/CassandraClusterInfoGroupTest.java

```diff
@@ -0,0 +1,222 @@
+        .hasMessage("Clusters are not running with the same partitioner kind. Found partitioners: " +
+                    "{cluster0=org.apache.cassandra.dht.Murmur3Partitioner, cluster1=org.apache.cassandra.dht.RandomPartitioner}");
+
+        CassandraClusterInfoGroup goodGroup = mockClusterGroup(2, index -> {
+            CassandraClusterInfo clusterInfo = mockClusterInfo("cluster" + index);
+            when(clusterInfo.getPartitioner()).thenReturn(Partitioner.Murmur3Partitioner);
+            return clusterInfo;
+        });
+        assertThat(goodGroup.getPartitioner()).isEqualTo(Partitioner.Murmur3Partitioner);
+    }
+
+    @Test
+    void testAggregateWriteAvailability()
+    {
+        Map<RingInstance, WriteAvailability> cluster1Availability = ImmutableMap.of(mock(RingInstance.class), WriteAvailability.AVAILABLE);
+        Map<RingInstance, WriteAvailability> cluster2Availability = ImmutableMap.of(mock(RingInstance.class), WriteAvailability.AVAILABLE);
+        CassandraClusterInfoGroup group = mockClusterGroup(2, index -> {
+            Map<RingInstance, WriteAvailability> availability = (index % 2 == 0) ? cluster1Availability : cluster2Availability;
+            CassandraClusterInfo clusterInfo = mockClusterInfo("cluster" + index);
+            when(clusterInfo.clusterWriteAvailability()).thenReturn(availability);
+            return clusterInfo;
+        });
+        assertThat(group.clusterWriteAvailability())
+        .describedAs("clusterWriteAvailability retrieved from group contains entries from both clusters")
+        .hasSize(2)
+        .containsValues(WriteAvailability.AVAILABLE, WriteAvailability.AVAILABLE);
+    }
+
+    @Test
+    void testAggregateLowestCassandraVersion()
+    {
+        CassandraClusterInfoGroup goodGroup = mockClusterGroup(2, index -> {
+            CassandraClusterInfo clusterInfo = mockClusterInfo("cluster" + index);
+            when(clusterInfo.getLowestCassandraVersion()).thenReturn("4.0." + index);
+            return clusterInfo;
+        });
+        assertThat(goodGroup.getLowestCassandraVersion()).isEqualTo("4.0.0");
+    }
+
+    @Test
+    void testAggregateLowestCassandraVersionFailDueToDifference()
+    {
+        CassandraClusterInfoGroup badGroup = mockClusterGroup(2, index -> {
+            CassandraClusterInfo clusterInfo = mockClusterInfo("cluster" + index);
+            when(clusterInfo.getLowestCassandraVersion()).thenReturn((4 + index) + ".0.0");
+            return clusterInfo;
+        });
+        assertThatThrownBy(badGroup::getLowestCassandraVersion)
+        .isExactlyInstanceOf(IllegalStateException.class)
+        .hasMessage("Cluster versions are not compatible. lowest=4.0.0 and highest=5.0.0");
+    }
+
+    @Test
+    void testCheckBulkWriterIsEnabledOrThrow()
+    {
+        CassandraClusterInfoGroup badGroup = mockClusterGroup(2, index -> {
+            CassandraClusterInfo clusterInfo = mockClusterInfo("cluster" + index);
+            if (index == 0)
```

Comment:
Similar to the "make sure I have different values", it would be good to test a non-zero index here (or randomize the index you use) - otherwise, aggregating this information and just calling the `checkBulkWriterIsEnabledOrThrow()` method would both pass the test.